src/status_im/contexts/centralized_metrics/tracking.cljs

```diff
@@ -4,12 +4,20 @@
     [react-native.platform :as platform]))
 
 (defn key-value-event
-  [event-name val-key value]
-  {:metric
-   {:eventName  event-name
-    :platform   platform/os
-    :appVersion build/app-short-version
-    :eventValue {val-key value}}})
+  [event-name & args]
+  (if (odd? (count args))
+    (throw (ex-info "Expected even number of args"
+                    {:args   args
+                     :n-args (count args)}))
+    {:metric
+     {:eventName  event-name
+      :platform   platform/os
+      :appVersion build/app-short-version
+      :eventValue (apply hash-map args)}}))
+
+(defn key-value-event-from-map
+  [event-name data]
+  (apply key-value-event event-name (flatten (seq data))))
```

Comment:
Side note:
I think we can possibly just re-write `key-value-event` to just take a map of arguments since I think the function is only used in this file roughly 3 times.